app/src/components/SignupForm.js

```diff
@@ -0,0 +1,80 @@
+import * as React from "react";
+
+import { useFormik } from "formik";
+import * as Yup from "yup";
+
+//Formik
+
+const SignupForm = () => {
+  const formik = useFormik({
+    initialValues: {
+      firstName: "",
+      lastName: "",
+      email: "",
+    },
+    validationSchema: Yup.object({
+      firstName: Yup.string()
+        .max(15, "Must be 15 characters or less")
+        .required("Required"),
+      lastName: Yup.string()
+        .max(20, "Must be 20 characters or less")
+        .required("Required"),
+      email: Yup.string().email("Invalid email address").required("Required"),
```

Comment:
This validation is really neat!!! 😎 Plus the form looks very sleek. It correctly doesn't allow emails like "avery@.com" or such.